src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
```

Comment:
What prevents us from just using the desktop code as-is?  Are we still missing some Assembly-related methods it depends on?